docs/DeveloperGuide.md

```diff
@@ -1,11 +1,119 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
+The Architecture Diagram above gives an overview of the different components in the application. 
+Details of the individual components are given below.  
+
+Mojo class controls the overall logic for the application. 
+It is responsible for 
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of 4 components:
+* Login: Links the user to their account.
+* Email: Stores and manages the user's email.
+* Utilities: Consists of Ui, Storage and Parser helper class. Ui interacts with the user. Storage reads data from, and writes data to the hard disk. Parser parse user's input into instructions that can be understood by the application.
+* Command: In charge of executing the user's request.
+
+For these four components,
+* The Login component exposes its functionality through the LoginController class.
+* The Email component exposes its functionality through the EmailManager class
+* The Utilities component consists of Ui, Storage and Parser classes.
+* The Command component exposes its functionality through different command classes that inherit from a base Command class.
+
+**How the architecture components interact with each other**
+![Architect sequence diagram](Sequence Diagrams/OverviewSequenceDiagram.png)
+The Sequence Diagram above shows how the components interact with each other for a command.
+1. `Mojo` uses the `LoginController` to get the user's userID and password.
+2. `Mojo` sends the user's info to the `Storage`, which will retrieve the existing emails from hard disks.
+3. `Mojo` updates the `EmailManager` with the retrieved emails
+4. `Mojo` gets the user's input from `Ui`
+5. `Mojo` sends the input to `Parser` which then return the respective `Command` class according to user's input
+6. `Mojo` executes the command.
+
+***Due to a limitation of PlantUML, the lifeline did not end at the end of the destroy marker**
+
+### Utilities Component
+The Utilities component contains the main classes that run the main functions of Mojo.
+
+![Utility diagram](UML diagrams/Utilities.jpeg)
```

Comment:
![Screenshot 2021-04-03 at 12 48 47 PM](https://user-images.githubusercontent.com/65282395/113468790-f018a500-947a-11eb-8b57-ab04c4028cd2.png)

Should there be 2 "throws" in the class diagram? Also the font might be a bit small and hard for some people to read.